config/routes.rb

```diff
@@ -21,10 +21,15 @@
         resources :merchant, only: :index
         resources :transactions, only: :index
       end
-      resources :invoice_items, only: [:index, :show] do
-        resources :invoice, only: :index
-        resources :item, only: :index
+
+      namespace :invoice_items do
+        get "/find", to: "finders#show"
+        get "/find_all", to: "finders#index"
+        get "/:id/invoice", to: "invoice#show"
```

Comment:
I think this /:id/ route would belong better nested under the invoice_items resources instead of namespace.